gdscript-comint.el

```diff
@@ -58,8 +58,17 @@
 ARGUMENTS are command line arguments for godot executable.
 When run it will kill existing process if one exists."
   (let ((buffer-name (gdscript-util--get-godot-buffer-name (member "-e" arguments)))
-        (inhibit-read-only t))
-    (if (not (or (file-executable-p gdscript-godot-executable) (executable-find gdscript-godot-executable)))
+        (inhibit-read-only t)
+        (godot-command gdscript-godot-executable))
+
+    ;; support flatpak
+    (if (string-match-p (rx "run" (* space) "org.godotengine.Godot") godot-command)
+        (let ((exec-split (split-string godot-command)))
+          (setq godot-command (car exec-split))
+          (setq arguments (append (cdr exec-split) arguments ))))
```

Comment:
We can use `when` here.

Make sure to remove trailing space after `arguments`.